### PR TITLE
MACOSX/IOS: Various updates from PR #1128

### PR DESCRIPTION
**Pull request by ccawley2011**
Originally from PR #1296, and modified to keep the iOS 3-6 packaging rules.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -2564,9 +2564,7 @@ case $_host_os in
 		esac
 
 		append_var DEFINES "-DMACOSX"
-		append_var LIBS "-framework AudioUnit -framework AudioToolbox -framework Carbon -framework CoreMIDI"
-		# SDL2 doesn't seem to add Cocoa for us.
-		append_var LIBS "-framework Cocoa"
+		append_var LIBS "-framework AudioUnit -framework AudioToolbox -framework Cocoa -framework CoreMIDI"
```
Comment (criezy) [inline configure]: SDL 1.2 needs Carbon, so we cannot just blindly remove it here. Actually SDL2 also still depends on it as well. But interestingly I do seem to get all of those when I use `sdl2-config --static-libs`, which is used in the `ports/mk` file. So I am not actually sure we need any of these here.

I would like some more time to check this.

**Comment by bluegr** [comment]
Any news on this?

**Comment by criezy** [comment]
I have merged the uncontroversial commits manually.

I don't want to merge commit 80bc3ae as is, and as a result I cannot merge commits 2c44681 and d0169e7. However those commits contain some nice change, but I will need more time to try to extract them if possible. I made a note of that and I am closing this PR now.

**Comment by bluegr** [comment]
Thanks :)

**Pull request closed without merging**